### PR TITLE
PWX-28082: Add support for AWS Secrets Manager

### PR DESCRIPTION
**Pull request by adityadani**
**What this PR does / why we need it**:
- Refactor the existing aws package to incorporate both KMS and Secrets Manager providers.
  - Moved the common code into a utils package.
- AWS Secrets Manager:
  - Implement Get/Put/Delete APIs.
  - Add integration tests for these APIs.

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:
Manually tested by running the integration tests with AWS credentials. Trying to see if we can hook that into travis.

### CODE REVIEW

File: aws/aws_secrets_manager/aws_scm.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+		}
+	}
+
+	id, secret, token, err := utils.AuthKeys(secretConfig)
+	if err != nil {
+		return nil, err
+	}
+	asc, err := sc.NewAWSCredentials(id, secret, token, true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get credentials: %v", err)
```
Comment (dahuang-purestorage) [inline aws/aws_secrets_manager/aws_scm.go:59]: nit: should the error message be more related to failing to form AWS credentials? 
maybe something like "failed to form AWS credentials: %v"

File: aws/aws_secrets_manager/aws_scm.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+		})
+	} else {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == secretsmanager.ErrCodeResourceNotFoundException {
+				// Create a new secret
+				_, err = a.scm.CreateSecret(&secretsmanager.CreateSecretInput{
+					SecretString: aws.String(string(secretBytes)),
+					Name:         aws.String(secretId),
+				})
+			} // return the aws error
```
Comment (dahuang-purestorage) [inline aws/aws_secrets_manager/aws_scm.go:142]: are we returning error here if it's not ResourceNotFoundException?

Comment (dahuang-purestorage) [reply aws/aws_secrets_manager/aws_scm.go:142]: oh nvm, i see the logic here. The error are propagated down to line 145.

Comment (dahuang-purestorage) [reply aws/aws_secrets_manager/aws_scm.go:142]: should we remove the comment then?

**Review by dahuang-purestorage — APPROVED** [review]
LGTM. just minor comments

File: aws/aws_secrets_manager/aws_scm.go (hunk 1 above)
Comment (adityadani) [reply aws/aws_secrets_manager/aws_scm.go:59]: makes sense. Will fix the error message

File: aws/aws_secrets_manager/aws_scm.go (hunk 2 above)
Comment (adityadani) [reply aws/aws_secrets_manager/aws_scm.go:142]: The comment right after an if closing statement indicates that the else condition will return back the error by this function

**Pull request merged**